internal/worker/bootstrap/worker.go

```diff
@@ -202,20 +253,20 @@ func (w *bootstrapWorker) seedAgentBinary(ctx context.Context, dataDir string) (
 	return cleanup, nil
 }
 
-func (w *bootstrapWorker) seedControllerCharm(ctx context.Context, dataDir string) error {
+func (w *bootstrapWorker) seedControllerCharm(ctx context.Context, dataDir string) (instancecfg.StateInitializationParams, error) {
```

Comment:
Just move `bootstrapParams` out of `seedControllerCharm`.

Reply:
done